sklearn/preprocessing/target.py

```diff
@@ -0,0 +1,222 @@
+# Authors: Guillaume Lemaitre <guillaume.lemaitre@inria.fr>
+#          Andreas Mueller <amueller@ais.uni-bonn.de>
+# License: BSD 3 clause
+
+import numpy as np
+
+from ..base import BaseEstimator, RegressorMixin, clone, is_regressor
+from ..linear_model import LinearRegression
+from ..utils.validation import check_is_fitted
+from ._function_transformer import FunctionTransformer
+
+__all__ = ['TransformedTargetRegressor']
+
+
+class TransformedTargetRegressor(BaseEstimator, RegressorMixin):
+    """Meta-estimator to apply a transformation to the target before fitting.
+
+    Useful for applying a non-linear transformation in regression
+    problems. This transformation can be given as a Transformer such as the
+    QuantileTransformer or as a function and its inverse such as ``np.log`` and
+    ``np.exp``.
+
+    The computation during ``fit`` is::
+
+        estimator.fit(X, func(y))
+
+    or::
+
+        estimator.fit(X, transformer.transform(y))
+
+    The computation during ``predict`` is::
+
+        inverse_func(estimator.predict(X))
+
+    or::
+
+        transformer.inverse_transform(estimator.predict(X))
+
+    Parameters
+    ----------
+    estimator : object, (default=LinearRegression())
+        Estimator object derived from ``RegressorMixin``.
+
+    transformer : object, (default=None)
+        Estimator object derived from ``TransformerMixin``. Cannot be set at
+        the same time as ``func`` and ``inverse_func``. If ``None`` and
+        ``func`` and ``inverse_func`` are ``None`` as well, the transformer
+        will be an identity transformer.
+
+    func : function, (default=None)
```

Comment:
Just sayin', if this says "optional" instead of "default=None", the docstring below should say "if not passed" instead of "If ``None``". (or the reader needs to scroll and check that the default value is None)